libs/community/langchain_community/vectorstores/yellowbrick.py

```diff
@@ -124,51 +211,69 @@ def add_texts(
         metadatas: Optional[List[dict]] = None,
         **kwargs: Any,
     ) -> List[str]:
-        """Add more texts to the vectorstore index.
-        Args:
-            texts: Iterable of strings to add to the vectorstore.
-            metadatas: Optional list of metadatas associated with the texts.
-            kwargs: vectorstore specific parameters
-        """
-        from psycopg2 import sql
+        batch_size = 10000
 
         texts = list(texts)
-        cursor = self._connection.cursor()
         embeddings = self._embedding.embed_documents(list(texts))
         results = []
         if not metadatas:
             metadatas = [{} for _ in texts]
-        for id in range(len(embeddings)):
-            doc_uuid = uuid.uuid4()
-            results.append(str(doc_uuid))
-            data_input = [
-                (str(id), embedding_id, text, json.dumps(metadata), embedding)
-                for id, embedding_id, text, metadata, embedding in zip(
-                    repeat(doc_uuid),
-                    range(len(embeddings[id])),
-                    repeat(texts[id]),
-                    repeat(metadatas[id]),
-                    embeddings[id],
-                )
-            ]
-            flattened_input = [val for sublist in data_input for val in sublist]
-            insert_query = sql.SQL(
-                "INSERT INTO {t} \
-                (id, embedding_id, text, metadata, embedding) VALUES {v}"
-            ).format(
-                t=sql.Identifier(self._table),
-                v=(
-                    sql.SQL(",").join(
-                        [
-                            sql.SQL("(%s,%s,%s,%s,%s)")
-                            for _ in range(len(embeddings[id]))
-                        ]
-                    )
-                ),
-            )
-            cursor.execute(insert_query, flattened_input)
-            self._connection.commit()
-        return results
+
+        conn = self._get_connection()
```

Comment:
Why the change to remove `INSERT INTO` behavior?

Reply:
It's significantly faster to use COPY to insert data than SQL INSERT.